### DISCUSSION TITLE
How can I know that the "Awaiting state lifecycle" is over

### DISCUSSION THREAD

**Discussion by just-website** [visgl/deck.gl#9262] [Q&A]
Deck ver. 9.0.35
I use deck and Mapbox Overlay.
I want to bind my app strore state and inner state deckGL as early as possible.
I try to call `myCompositeLayer.setState(newState)` inside `map.onLoad` callback. And even though the Deck layers are ready, I get an error `'Cannot convert undefined or null to object'`. As I see it, this is because the internal `_initialize()` method has not yet been called from the LayerManager. 
Is there an event after which `layer.setState()` can be called the first time?

![image](https://github.com/user-attachments/assets/dd9f7568-03eb-49d8-bd90-fb85dcafdf28)

**Comment by Pessimistress** [comment]
`layer.setState` is not supposed to be called from outside of the layer (unless you know what you are doing).
